contracts/contracts/harvest/Harvester.sol

```diff
@@ -0,0 +1,376 @@
+     */
+    function removeSwapToken(address _addr) external onlyGovernor {
+        uint256 swapTokenIndex = swapTokens.length;
+        for (uint256 i = 0; i < swapTokens.length; i++) {
+            if (swapTokens[i] == _addr) {
+                swapTokenIndex = i;
+                break;
+            }
+        }
+
+        require(swapTokenIndex != swapTokens.length, "Swap token not added");
+
+        // Shift everything after the index element by 1
+        for (uint256 i = swapTokenIndex; i < swapTokens.length - 1; i++) {
+            swapTokens[i] = swapTokens[i + 1];
+        }
+        swapTokens.pop();
+
+        emit SwapTokenRemoved(_addr);
+    }
+
+    /**
+     * @dev Add/update a reward token configuration that holds harvesting config variables
+     * @param _tokenAddress Address of the reward token
+     * @param _allowedSlippageBps uint16 maximum allowed slippage denominated in basis points. Example: 300 == 3% slippage
+     * @param _harvestRewardBps uint16 amount of reward tokens the caller of the function is rewarded. Example: 100 == 1%
+     * @param _uniswapV2CompatibleAddr Address Address of a UniswapV2 compatible contract to perform the exchange from reward
+     *        tokens to stablecoin (currently hard-coded to USDT)
+     * @param _liquidationLimit uint256 Maximum amount of token to be sold per one swap function call. When value is 0 there is no limit.
+     */
+    function setRewardTokenConfig(
+        address _tokenAddress,
+        uint16 _allowedSlippageBps,
+        uint16 _harvestRewardBps,
+        address _uniswapV2CompatibleAddr,
+        uint256 _liquidationLimit
+    ) external onlyGovernor {
+        require(
+            _harvestRewardBps <= 1000,
+            "Harvest reward fee should not be over 10%"
+        );
+        require(
+            _allowedSlippageBps <= 1000,
+            "Allowed slippage should not be over 10%"
+        );
+        require(
+            _uniswapV2CompatibleAddr != address(0),
+            "Uniswap compatible address should be non zero address"
+        );
+
+        RewardTokenConfig memory tokenConfig = RewardTokenConfig({
+            harvestRewardBps: _harvestRewardBps,
+            allowedSlippageBps: _allowedSlippageBps,
+            uniswapV2CompatibleAddr: _uniswapV2CompatibleAddr,
+            liquidationLimit: _liquidationLimit
+        });
+
+        address oldUniswapAddress = rewardTokenConfigs[_tokenAddress]
+            .uniswapV2CompatibleAddr;
+        rewardTokenConfigs[_tokenAddress] = tokenConfig;
+
+        IERC20 token = IERC20(_tokenAddress);
+
+        // if changing token swap provider cancel existing allowance
+        if (
+            oldUniswapAddress != address(0) &&
+            oldUniswapAddress != _uniswapV2CompatibleAddr
+        ) {
+            token.safeApprove(oldUniswapAddress, 0);
+        }
+
+        // Give Uniswap infinite approval when needed
+        if (_uniswapV2CompatibleAddr != address(0)) {
+            if (
```

Comment:
I think it's more efficient/cleaner to only set allowances if we are changing to a new uniswap address.

2^256 an an allowance is stupidly big number. We shouldn't run it out.

Reply:
cool, agreed 👍